book/src/article/generate_multiple_files.md

```diff
@@ -183,4 +183,74 @@ gen:
+
+Note: For a shared used customized type(struct/enum), DON'T define it in ANY regular Api definition file(referred by flag `rust-input`). Because it is not a good practice, also it would generate issued Rust files.
+
+
+### Shared Dart file
+
+For the generated Dart files, all the generated stuff is just like the Rust files. And to use a customized type that is shared, just like for the regular one, the shared Api object needs to be defined, like this:
+```dart
+// This file initializes the dynamic library and connects it with the stub
+// generated by flutter_rust_bridge_codegen.
+
+import 'dart:ffi';
+import 'dart:io' as io;
+
+import 'gened_api_1.dart';
+import 'gened_api_2.dart';
+
+const _base = 'native';
+
+// On MacOS, the dynamic library is not bundled with the binary,
+// but rather directly **linked** against the binary.
+final _dylib = io.Platform.isWindows ? '$_base.dll' : 'lib$_base.so';
+
+final api1 = Api1Impl(io.Platform.isIOS || io.Platform.isMacOS
+    ? DynamicLibrary.executable()
+    : DynamicLibrary.open(_dylib));
+
+final api2 = Api2Impl(io.Platform.isIOS || io.Platform.isMacOS
+    ? DynamicLibrary.executable()
+    : DynamicLibrary.open(_dylib));
+
+final apiShare = BridgeGeneratedSharesImpl(
+    io.Platform.isIOS || io.Platform.isMacOS
+        ? DynamicLibrary.executable()
+        : DynamicLibrary.open(_dylib));
+
+```
+Here, `apiShare` is responsible for any shared types. For example, if there is a shared type named `SharedStructInBlock1And2`, now it can be used with `bridge` as the shared Api object, like this:
+```dart
+    var obj = SharedStructInBlock1And2(bridge: apiShared, name: "newString", id: 2, num: 2.2);
+    // using methods
+    expect("msg_1", await obj.testMethod(message: "msg", num: 1));
+    // using static methods directly by class name
+    expect("msg", await SharedStructInBlock1And2.testStaticMethod(bridge: api1, message: "msg"));
+```
+
+### Shared C header file
+
+Besides, for the generated C (header) files, like the corresponding generated Rust and Dart files, there would be also an auto-generated C file with the name `bridge_generated_shares.h`(the name would also be changed by flag `shared_rust_output`) besides the other C files for regular blocks.
+
+This shared C file, besides all shared types and Apis, also includes ALL the headers of all regular C files. That is, it could be treated as the manager of all C files.
```

Comment:
I am not sure whether this is the best way. On one hand, it seems we put "common utility" into the shared.rs/c/dart file, i.e. usually it is block_1.rs/c/dart imports shared.rs/c/dart; but on the other hand, here we want to make it the file that includes other files.

One way may be, create another header file, say `bridge_generated_umbrella.h` (I temporarily call it umbrella because the semantics is a bit like https://stackoverflow.com/questions/31238761/what-is-an-umbrella-header). This file has nothing but :

```c
#include "bridge_generated_shared.h"
#include "bridge_generated_block_1.h"
#include "bridge_generated_block_2.h"
```

i.e. it includes everything (and do not have content on its own).

Then, users only need to include this file, and done.

Or, we may even name it `bridge_generated.h` (with the same content above - just a few #include). Then, our tutorial will be exactly the same for one-block and multi-block, since the users only need to worry about bridge_generated.h and no else.

Please correct me if I am wrong!